warehouse/utils/context.go

```diff
@@ -0,0 +1,20 @@
+package warehouseutils
+
+import (
+	"context"
+)
+
+type key int
+
+const (
+	uploadID key = iota
```

Comment:
[optional] The code is correct but I suggest using `struct` types to handle context keys since `uploadID` can too easily be converted by someone to `int` opening this code to unintentional overwrites of the `uploadID`.

```go
type uploadIDContextKey struct {}

func CtxWithUploadID(ctx context.Context, uid int64) context.Context {
	return context.WithValue(ctx, uploadIDContextKey{}, uid)
}

func UploadIDFromCtx(ctx context.Context) (int64, bool) {
	userIP, ok := ctx.Value(uploadIDContextKey{}).(int64)
	return userIP, ok
}
```

Less lines of code too :stuck_out_tongue: